backends/decorators/metrics.go

```diff
@@ -10,49 +10,47 @@ import (
 )
 
 type backendWithMetrics struct {
-	delegate backends.Backend
-	puts     *metrics.MetricsEntryByFormat
-	gets     *metrics.MetricsEntry
+	delegate       backends.Backend
+	metricsEngines *metrics.Metrics
 }
 
 func (b *backendWithMetrics) Get(ctx context.Context, key string) (string, error) {
-	b.gets.Request.Mark(1)
+	b.metricsEngines.RecGetBackendRequest("add", nil)
```

Comment:
It'll add '1' to the influx or prometheus counter that keeps track of the number of backend requests. In terms of Influx, this means we add one to the influx metrics like: `InfluxMetrics.GetsBackend.Request.Mark(1)` and/or the prometheus metrics like: `PrometheusMetrics.GetsBackend.RequestStatus.With(prometheus.Labels{"status":"add"}).Inc()`

Reply:
I can change the keyword name from "add" to "increment", "mark", or something else

Reply:
I'm inferring that "add" = request begin, "" = request complete successfully, and "error" = request completed with error. If so, consider splitting the methods to `RecordGetBackendRequestBegin()` and `RecordGetBackendRequestFinish(err)`.

Reply:
In this part of the code I just went with the logic that was already there for Influx and function `RecGetBackendRequest(status string, duration *time.Time)` is meant to record metrics, on whatever engine we use, that relate to the "GET" requests on the backend. This function was meant to be executed like:

- `RecGetBackendRequest("add", nil)` - Will increase the counter of the get backend requests whether this request will be successful or not. It's meant to execute `b.gets.Request.Mark(1)` on influx, and `m.GetsBackend.RequestStatus.With(prometheus.Labels{label: {"add"})` on the Prometheus engine.
- `RecGetBackendRequest("", &start)` - This was a successful request for which we have an execution duration in the `&start` parameter. It's meant to execute `b.gets.Duration.UpdateSince(start)` on influx, and `m.GetsBackend.Duration.Observe(time.Since(*duration).Seconds())` on the Prometheus engine.
- `RecGetBackendRequest("error", nil)` - This was an unsuccessful request for which we have an execution duration but we don't record it, we just add to the vector tracking the number of requests that came with an error. It's meant to execute `b.gets.Errors.Mark(1)` on influx, and `m.GetsBackend.RequestStatus.With(prometheus.Labels{label: {"error"})` on the Prometheus engine.

I realize that this can be confusing. In reference to your other comment, I'll create more functions so we separate these three functionalities, and will come up with names that are more intuitive